packages/web/pages/dashboard.tsx

```diff
@@ -10,32 +9,25 @@ import {
   Player,
   useUpdatePlayerDashboardLayoutMutation as useUpdateLayout,
 } from 'graphql/autogen/types';
+import { useUser } from 'lib/hooks';
 import React, { useCallback, useMemo } from 'react';
 import { LayoutData } from 'utils/boxTypes';
 
-const ConnectedDashboardPage: React.FC<Props> = () => (
-  <ConnectedPage page={DashboardPage} pageLabel="Your Dashboard" />
-);
-
-export default ConnectedDashboardPage;
-
-type Props = { player: Player };
-
-export const DashboardPage: React.FC<Props> = ({ player }) => {
+const DashboardPage: React.FC = () => {
   const [{ fetching: persisting }, saveLayoutData] = useUpdateLayout();
-
+  const { user: player } = useUser();
   const savedLayoutData = useMemo<LayoutData>(
     () =>
-      player.dashboardLayout
-        ? JSON.parse(player.dashboardLayout)
+      player?.dashboardLayout
+        ? JSON.parse(player?.dashboardLayout)
         : DEFAULT_DASHBOARD_LAYOUT_DATA,
-    [player.dashboardLayout],
+    [player?.dashboardLayout],
   );
 
   const persistLayoutData = useCallback(
     async (layoutData: LayoutData) => {
       const { error } = await saveLayoutData({
-        playerId: player.id,
+        playerId: player?.id,
```

Comment:
Rather than trying to call `saveLayoutData` with an `undefined` `playerId` *(which is going to error)*, you probably should only call `saveLayoutData` if `player` is defined.

Reply:
resolved